frontend/awx/views/jobs/WorkflowOutputNavigation.tsx

```diff
@@ -0,0 +1,139 @@
+import { useState, useMemo } from 'react';
+import {
+  MenuToggle,
+  MenuToggleElement,
+  Select,
+  SelectOption,
+  SelectGroup,
+  SelectList,
+} from '@patternfly/react-core';
+import { CheckCircleIcon, ExclamationCircleIcon } from '@patternfly/react-icons';
+import { Link, useParams } from 'react-router-dom';
+import { useTranslation } from 'react-i18next';
+import { stringIsUUID } from '../../common/util/strings';
+import { jobPaths } from './WorkflowOutput/WorkflowOutputNode';
+import type { WorkflowJobNode } from '../../interfaces/WorkflowNode';
+
+interface WorkflowOutputNavigationProps {
+  workflowNodes: WorkflowJobNode[];
+}
+
+export function WorkflowOutputNavigation(props: WorkflowOutputNavigationProps) {
+  const { t } = useTranslation();
+  const { id } = useParams<{ id: string }>();
+  const { workflowNodes } = props;
+
+  const [isOpen, setIsOpen] = useState<boolean>(false);
+  const [selected, setSelected] = useState<string | undefined>();
+  const [filteredNodes, setFilteredNodes] = useState<WorkflowJobNode[]>(workflowNodes);
+
+  const totalResults = workflowNodes.length || 0;
+
+  const statuses: Record<string, string[]> = useMemo(
+    () => ({
+      failed: ['error', 'failed'],
+      successful: ['successful'],
+    }),
+    []
+  );
+
+  const flatStatuses = useMemo(
+    () => Object.values(statuses).reduce((acc, cur) => [...acc, ...cur], []),
+    [statuses]
+  );
+
+  const onToggleClick = () => {
+    setIsOpen(!isOpen);
+  };
+
+  const onSelect = (event: React.MouseEvent | undefined, v: string | number | undefined) => {
+    if (typeof v === 'undefined') return;
+
+    const value = v.toString();
+
+    if (value === selected) {
+      setSelected(undefined);
+      setFilteredNodes(workflowNodes);
+    } else {
+      setSelected(value);
+
+      if (!flatStatuses.includes(value)) return;
+
+      setFilteredNodes(
+        workflowNodes.filter(
+          (node) =>
+            node.summary_fields?.job?.status &&
+            statuses[value].includes(node.summary_fields?.job?.status) &&
+            node.job?.toString() !== id
+        )
+      );
+    }
+  };
+
+  const Toggle = (toggleRef: React.Ref<MenuToggleElement>) => (
+    <MenuToggle
+      ref={toggleRef}
+      onClick={onToggleClick}
+      isExpanded={isOpen}
+      style={
+        {
+          width: '180px',
+        } as React.CSSProperties
+      }
+    >
+      {t(`Workflow Job 1/{{totalResults}}`, { totalResults })}
+    </MenuToggle>
+  );
+
+  return (
```

Comment:
@jerabekjiri Good job. Would be possible to use the components provided the framework?

Reply:
hmm... that could be a good idea, I'll test that

Reply:
@nixocio so I have tried to re-implement with the `PageSingleSelect` component, but I'm unable to control the component and keep it open while filtering since this component works more like a filter than normal select